### DISCUSSION TITLE
Support for multiple accounts in Exelon companies (BGE)

### DISCUSSION THREAD

**Discussion by cz-unit** [tronikos/opower#98] [Q&A]
Hi! Just fired this up in Home Assistant and noticed that it only can find the elements for the first account at my house (I have two meters, 2 meter accounts, one bge.com account). Is it possible or is the code supposed to find all accounts? If not can an account be selected for importing into HA?

I can see in exelon.py that it does pick an account if one doesn't pop up, maybe I'll start there by forcing a different account number. But that's a seriously bad hack :-)

**Comment by apopticon** [comment]
Did you have any luck with this? I have two subaccounts on my BGE account, and Opower grabs the first one, which of course is the wrong one.

**Comment by apopticon** [comment]
Wonder if changing exelon.py line 93 from `return active_accounts[0]` to return `active_accounts[1]` would get me what I need...  May mess with this later.

**Reply by cz-unit** [reply]
Yep, that can do it. Odly enough it can take a day or two before the code is "run" to update the other account. Give it a go and let us know how it works for you (I believe that's what did it for me).

Oddly enough I put in a line in my notes saying "This is putting the values into the database, where I can see it, but is not properly updating the tiles or the objects. I’ll have to fix that later, but in the meantime I can just switch the value from 1 to zero every month or so to get all the data downloaded properly."

So, a kludge, but it works. Let me know if you get further.

**Reply by apopticon** [reply]
I logged into the host as root and exec'ed a shell in the HA container, and edited /usr/local/lib/python3.13/site-packages/opower/utilities/exelon.py right there. Doesn't seem to work as hoped. Adding the entry completes with no error, and from the logfile, it says it downloaded a bunch of data, but there is nothing usable.  If I delete the entry, change the subscript back to 0 and add the entry, I immediately get the data from the 1st account.  I didn't wait a day or two, suppose I could try that... 

I have no idea what the data format returned from opower looks like, it was a shot in the dark.  Increasing the subscript past 1 gets an error, so there's no more after that.

**Reply by apopticon** [reply]
Following up on my own reply; I deleted the integration, and then installed the latest HA update & restarted HA. Then edited the subscript in exelon.py, then added opower integration. This time, it seems to have reported the correct account number, but the data doesn't seem to have populated yet. Interestingly, the actual utility company account number isn't reported, it actually reports the "Electric Choice ID". I'll leave it a couple days and see if it ever updates.

**Reply by apopticon** [reply]
It took two days, but the data is now showing!

**Comment by cz-unit** [comment]
Weird, isn't it? Glad you too got this working, I need to learn how python's compilers and intermediate code works, also Opower is notorious for not updating for a week at a time. Not due to HA, just due to your power company (verified by logging into the power company's web site and not seeing the updates for days at a time)

Still, glad it helped you. May both of our posts on this thread help others.....
